internal/activity/pusher.go

```diff
@@ -0,0 +1,55 @@
+package activity
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"cdr.dev/coder-cli/internal/entclient"
+	"go.coder.com/flog"
+)
+
+const pushInterval = time.Minute
+
+// Pusher pushes activity metrics no more than once per pushInterval. Pushes
+// within the same interval are a no-op.
+type Pusher struct {
+	Source string
+	EnvID  string
+	Client *entclient.Client
+
+	state int64
+}
+
+func (p *Pusher) Push() {
+	if atomic.CompareAndSwapInt64(&p.state, 0, 1) {
+		err := p.Client.PushActivity(p.Source, p.EnvID)
+		if err != nil {
+			flog.Error("failed to push activity: %s", err.Error())
+		}
+	}
+}
+
+// Start starts the reset routine. It resets the state every pushInterval,
+// allowing Push to push a new activity.
+func (p *Pusher) Start() (end func()) {
```

Comment:
Usually, these are better structured when they take a context. The code ends up with less `WithCancel`s and calls to `cancel()`.